Error: An error was encountered while compiling documentation

### Description

```
CompileDocumentation /Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax/Sources/SwiftBasicFormat/SwiftBasicFormat.docc (in target 'SwiftBasicFormat' from project 'swift-syntax')
    cd /Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/docc convert --emit-lmdb-index --fallback-display-name SwiftBasicFormat --fallback-bundle-identifier swift-syntax.SwiftBasicFormat --fallback-bundle-version 0 --output-dir /Users/ec2-user/Library/Developer/Xcode/DerivedData/swift-syntax-doldviwbxbfivxgmqcmdlduayohg/Build/Products/Debug/SwiftBasicFormat.doccarchive --ide-console-output --diagnostics-file /Users/ec2-user/Library/Developer/Xcode/DerivedData/swift-syntax-doldviwbxbfivxgmqcmdlduayohg/Build/Intermediates.noindex/swift-syntax.build/Debug/SwiftBasicFormat.build/SwiftBasicFormat-diagnostics.json --warnings-as-errors /Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax/Sources/SwiftBasicFormat/SwiftBasicFormat.docc --additional-symbol-graph-dir /Users/ec2-user/Library/Developer/Xcode/DerivedData/swift-syntax-doldviwbxbfivxgmqcmdlduayohg/Build/Intermediates.noindex/swift-syntax.build/Debug/SwiftBasicFormat.build/symbol-graph
/Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax/Sources/SwiftBasicFormat/Indenter.swift:24:100: error: Parameter 'indentation' is missing documentation (in target 'SwiftBasicFormat' from project 'swift-syntax')
    note: Document 'indentation' parameter
/Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax/Sources/SwiftBasicFormat/Indenter.swift:24:100: error: Parameter 'indentation' is missing documentation. Document 'indentation' parameter.
/Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax/Sources/SwiftBasicFormat/Indenter.swift:22:7-22:7: fixit: - Parameter indentation: <#parameter description#>
///
Error: An error was encountered while compiling documentation
```

### Steps to Reproduce

Xcode 16.2 + macOS 15.2

1. on main branch 
2. /usr/bin/xcodebuild docbuild -workspace /Users/ec2-user/jenkins/workspace/oss-swift-pr-test-macos-arm64/swift-syntax -scheme SwiftBasicFormat -destination platform=macOS OTHER_DOCC_FLAGS='--warnings-as-errors'

[consoleText-48.txt](https://github.com/user-attachments/files/18729960/consoleText-48.txt)

--- a/SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/VerifyDocumentation.swift
+++ b/SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/VerifyDocumentation.swift
@@ -46,11 +46,34 @@ struct VerifyDocumentation: ParsableCommand {
     }
   }
 
+  func xcodeVersion(xcodebuildExec: URL) throws -> (major: Int, minor: Int) {
+    let result = try ProcessRunner(
+      executableURL: xcodebuildExec,
+      arguments: ["-version"]
+    ).run(captureStdout: true, captureStderr: false, verbose: false)
+    let xcodeVersionRegex = Regex {
+      "Xcode "
+      Capture(OneOrMore(.digit))
+      "."
+      Capture(OneOrMore(.digit))
+    }
+    guard let match = result.stdout.firstMatch(of: xcodeVersionRegex), let major = Int(match.1),
+      let minor = Int(match.2)
+    else {
+      throw ScriptExectutionError(message: "Failed to extract Xcode version to verify documentation")
+    }
+    return (major, minor)
+  }
+
   func buildDocumentation(product: String) throws {
     guard let xcodebuildExec = try? Paths.xcodebuildExec else {
       return
     }
     logSection("Building documentation for \(product)")
+    var otherDoccFlags = ["--warnings-as-errors"]
+    if try xcodeVersion(xcodebuildExec: xcodebuildExec) >= (16, 0) {
+      otherDoccFlags.append("--disable-parameters-and-returns-validation")
+    }
     try ProcessRunner(
       executableURL: xcodebuildExec,
       arguments: [
@@ -60,7 +83,7 @@ struct VerifyDocumentation: ParsableCommand {
         "-scheme",
         product,
         "-destination", "platform=macOS",
-        "OTHER_DOCC_FLAGS='--warnings-as-errors'",
+        "OTHER_DOCC_FLAGS=\(otherDoccFlags.joined(separator: " "))",
       ]
     ).run(captureStdout: false, captureStderr: false, verbose: self.verbose)
   }

